tmk_core/protocol/chibios/usb_report_handling.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Stefan Kerkmann (@KarlK90)
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <ch.h>
+#include <hal.h>
+#include <stdint.h>
+
+#include "timer.h"
+
+typedef struct {
+    time_msecs_t idle_rate;
+    systime_t    last_report;
+    uint8_t      data[64];
+    size_t       length;
+} usb_fs_report_t;
+
+typedef struct {
+    usb_fs_report_t **reports;
+    const void (*get_report)(usb_fs_report_t **, uint8_t, usb_fs_report_t *);
+    const void (*set_report)(usb_fs_report_t **, const uint8_t *, size_t);
+    const void (*reset_report)(usb_fs_report_t **);
+    const void (*set_idle)(usb_fs_report_t **, uint8_t, uint8_t);
+    const uint8_t (*get_idle)(usb_fs_report_t **, uint8_t);
+    const bool (*idle_timer_elasped)(usb_fs_report_t **, uint8_t);
+} usb_report_storage_t;
+
+#define QMK_USB_REPORT_STROAGE_ENTRY(_report_id, _report_size) [_report_id] = &((usb_fs_report_t){.data = {[0] = _report_id}, .length = _report_size})
+
+#define QMK_USB_REPORT_STORAGE(_get_report, _set_report, _reset_report, _get_idle, _set_idle, _idle_timer_elasped, _report_count, _reports...) \
+    &((usb_report_storage_t){                                                                                                                  \
+        .reports            = (_Alignas(4) usb_fs_report_t *[_report_count]){_reports},                                                        \
```

Comment:
It turned out that this kind of usage of `_Alignas` in compound literals is supported [only in GCC >= 8.x](https://gcc.gnu.org/legacy-ml/gcc-patches/2017-12/msg00406.html).  Because of that, GNU Guix is no longer suitable for building the QMK firmware (they have only `gcc-arm-none-eabi-7-2018-q2-update`).

This particular usage of `_Alignas` is also questionable — apparently it is applied to an array of pointers, which should be aligned appropriately anyway.  Other places where the alignment is applied to byte arrays used for endpoint buffers look reasonable though.